pkg/query-service/app/integrations/messagingQueues/kafka/sql.go

```diff
@@ -318,6 +318,97 @@ GROUP BY
 	return query
 }
 
+func generateOverviewSQL(start, end int64) string {
+	// Convert from nanoseconds to float seconds in Go
+	// to avoid decimal overflow in ClickHouse
+	startSeconds := float64(start) / 1e9
+	endSeconds := float64(end) / 1e9
+
+	// For timeRange, if you want the difference in the DB:
+	timeRangeSecs := endSeconds - startSeconds // float64 difference in Go
+
+	// If you really need tsBucketStart, do the math in Go as well
+	tsBucketStart := startSeconds - 1800 // example
+	tsBucketEnd := endSeconds
+
+	query := fmt.Sprintf(`
+WITH
+    timeRange AS (
+        SELECT %f AS seconds
+    ),
+    processed_traces AS (
+        SELECT
+            resource_string_service$$name AS service_name,
+            name AS span_name,
+            CASE
+                WHEN attribute_string_messaging$$system = 'kafka' THEN 'kafka'
+                WHEN (has(attributes_string, 'celery.action') OR has(attributes_string, 'celery.task_name')) THEN 'celery'
+                ELSE 'other'
+            END AS messaging_system,
```

Comment:
Can this be moved to write at collector and users can just rely on `attribute_string_messaging$$system` always?

Reply:
I am keeping it here since ideally upstream should implement this in the SDK, once it's is implemented correctly in upstream, we don't need this check, there's an issue open in upstream for that https://github.com/open-telemetry/opentelemetry-python-contrib/issues/2913

Reply:
this is how I am doing the filtering, https://github.com/SigNoz/signoz/pull/6756/files#diff-480b722b377e24e0d7eff30cee9640595400fd21ffbc67283b67ff7e78385815R324

Reply:
marking this as resolved